### PR TITLE
[fix]: [PL-30826]: Fixing issue where deleted permission not getting deleted from custom roles

### PR DESCRIPTION
**Pull request by ashishsanodia-harness**
## Describe your changes
Fixing issue where if we delete permission from permission.yml It was only getting deleted from managed roles but not from custom roles.

The issue was with criteria creation. When permission management job runs it passes managed filter as `NO_FILTER` it creates below criteria

> criteria.and(RoleDBOKeys.scopeIdentifier).in(roleFilter.getScopeIdentifier(), null);

from permission management job roleFilter.getScopeIdentifier() is coming as null with `NO_FILTER` so final criteria will have

> "scopeIdentifier":{ "$in" : [null, null]}

And it will always end up matching managed roles where scope identifier is null.

There are other flows(role listing) which are using this criteria creation and pass scopeIdentifier along with `NO_FILTER`

Fix:
In case of NO_FILTER 
- if scopeIdentifier is not present we should not add any criteria for scopeIdentifier so that it will consider all the roles MANAGED + CUSTOM without scopeIdentifer. 
- If scopeIdentifier is present we will add criteria for scopeIdentifier so that it will consider all the roles MANAGED + CUSTOM with scopeIdentifer

Tests:
- Verified If we mark permission `INACTIVE` in `permissions.yml` it is getting removed from `managed` roles as expected
- Verified If we mark permission `INACTIVE` in `permissions.yml` it is getting removed from `custom` roles as expected
- Verified If we `remove` permission from `permissions.yml` it is getting removed from `managed` roles as expected
- Verified If we `remove` permission from `permissions.yml `it is getting removed from `custom` roles as expected

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55784)

### CODE REVIEW

**Comment by ashishsanodia-harness** [comment]
trigger uts

**Comment by ashishsanodia-harness** [comment]
trigger messagecheck

**Comment by ashishsanodia-harness** [comment]
trigger uts

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55784)* status: 2 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/persistence/RoleDaoImpl.java` line 180 at r2](https://reviewable.io/reviews/harness/harness-core/55784#-NjpYo379HCVen33yV-s:-NjpYo379HCVen33yV-t:b8icxg4) ([raw file](https://github.com/harness/harness-core/blob/c0ad0e5a77e591834d1368bffd267377e81eee39/access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/persistence/RoleDaoImpl.java#L180)):*
> ```Java
>       criteria.and(RoleDBOKeys.scopeIdentifier).is(null);
>       criteria.and(RoleDBOKeys.managed).is(true);
>     } else if (roleFilter.getManagedFilter().equals(NO_FILTER)) {
> ```

Lets remove this If condition. It is self-explanatory.

**Comment by ashishsanodia-harness** [comment]
trigger uts

**Comment by ashishsanodia-harness** [comment]
*[`access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/persistence/RoleDaoImpl.java` line 180 at r2](https://reviewable.io/reviews/harness/harness-core/55784#-NjpYo379HCVen33yV-s:-NjpaH0w0CoOCS2IEyud:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/c0ad0e5a77e591834d1368bffd267377e81eee39/access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/persistence/RoleDaoImpl.java#L180)):*
<details><summary><i>Previously, Jimit09Gandhi (Jimit Gandhi) wrote…</i></summary><blockquote>

Lets remove this If condition. It is self-explanatory.
</blockquote></details>

Done.

**Review by Jimit09Gandhi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55784)* status: 2 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/persistence/RoleDaoImpl.java` line 180 at r2](https://reviewable.io/reviews/harness/harness-core/55784#-NjpYo379HCVen33yV-s:-NjpbpGO-c9W5f-A7wii:b-dpvi7f) ([raw file](https://github.com/harness/harness-core/blob/c0ad0e5a77e591834d1368bffd267377e81eee39/access-control/libs/core/src/main/java/io/harness/accesscontrol/roles/persistence/RoleDaoImpl.java#L180)):*
<details><summary><i>Previously, ashishsanodia-harness (Ashish Sanodia) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Should we do 
if(isNotNull(scopeIdentifier)
 {
criteria.and(RoleDBOKeys.scopeIdentifier).is(roleFilter.getScopeIdentifier());
}
OR criteria.and(RoleDBOKeys.scopeIdentifier).is(null);

basically it should't happen that Delete role called for specific identifier deletes roles of all other scopes. @ashishsanodia-harness Can u think about cases for same?

**Comment by ashishsanodia-harness** [comment]
trigger uts

**Comment by ashishsanodia-harness** [comment]
trigger uts
